src/cmap/mod.rs

```diff
@@ -57,38 +54,139 @@ impl From<Arc<ConnectionPoolInner>> for ConnectionPool {
     }
 }
 
-/// The internal state of a connection pool.
-#[derive(Derivative)]
-#[derivative(Debug)]
-pub(crate) struct ConnectionPoolInner {
-    /// The address the pool's connections will connect to.
-    address: StreamAddress,
-
+/// A struct used to manage the creation, closing, and storage of connections for a `ConnectionPool`.
+#[derive(Debug)]
+struct ConnectionManager {
```

Comment:
This type encapsulates all the connection related state that needs to be accessed from behind the lock. 
- As discussed offline, it `total_connection_count` should only be accessed if a write lock is held on the `Vec<Connections>`, so it makes sense to group them. 
- The same is generally true for `next_connection_id`. 
- Based on just looking at where we used `generation`, I saw that it was _only_ being used when the lock was held, so there was no cost to keeping it here with the rest of the state. The only exception is that `clear` would previously update the generation without getting the write lock on the connections, but the argument could be made that it should.

Reply:
I like this refactor a lot! The separation makes the pool a lot cleaner.

Just to make sure I understand, the Establisher is in here so that it can be more easily accessed by the background thread?